src/main/java/org/folio/circulation/domain/RequestServiceUtility.java

```diff
@@ -50,13 +52,23 @@ static Result<RequestAndRelatedRecords> refuseWhenRequestCannotBeFulfilled(
     RequestAndRelatedRecords requestAndRelatedRecords) {
 
     RequestPolicy requestPolicy = requestAndRelatedRecords.getRequestPolicy();
-    RequestType requestType = requestAndRelatedRecords.getRequest().getRequestType();
+    Request request = requestAndRelatedRecords.getRequest();
+    RequestType requestType = request.getRequestType();
 
     if (!requestPolicy.allowsType(requestType)) {
       return failureDisallowedForRequestType(requestType);
-    } else {
-      return succeeded(requestAndRelatedRecords);
     }
+
+    if (!requestPolicy.allowsServicePoint(requestType, request.getPickupServicePointId())) {
+      return failedValidation("One or more Pickup Locations are no longer available",
```

Comment:
I wonder if we should start collecting these messages in one place, maybe we should create a tech debt task for this. We've already been asked to provide all possible validation error messages once.

Reply:
Will create a ticket for that